tests/cli-commands/deploy-tests.js

```diff
@@ -0,0 +1,170 @@
+const DeployerOption = require('../../cli-commands/commands/deploy/options/deployer-option');
+
+describe('DeployCommand', function () {
+    const TEST_DIR = './cli-commands-test';
+    const DEFAULT_PATH = './deployment';
+    const INVALID_PATH = './unknown_folder';
+    const DEFAULT_NETWORK = 'local';
+    const INVALID_NETWORK = 'invalid_network';
+    const CUSTOM_NETWORK = { url: "https://test.custom.net", chainId: "cf057bbfb72640471fd910bcb67639c22df9f92470936cddc1ade0e2f2e7dc4f" };
+    const DEPLOYER_NAME = 'alice';
+    const DEPLOYER_PERMISSION = 'active';
+    const DEPLOYER_PRIVATE_KEY = '5JymWHYohPMXTckmmFkmiRZZDQTWN91eDFSnEvjYAbXTd6oFtie';
+    const INVALID_PRIVATE_KEY = 'invalid_private_key';
+
+    let initialDir;
+    let deployCommand;
+    let eoslimeSpy;
+    let pathOptionSpy;
+    let networkOptionSpy;
+    let deployerOptionSpy;
+
+    before(async () => {
+        initialDir = process.cwd();
+        fs.mkdirSync(TEST_DIR);
+        process.chdir(TEST_DIR);
+    });
+
+    beforeEach(async () => {
+        deployCommand = new DeployCommand();
+        eoslimeSpy = sinon.spy(eoslime, "init");
+        pathOptionSpy = sinon.spy(PathOption, "process");
+        networkOptionSpy = sinon.spy(NetworkOption, "process");
+        deployerOptionSpy = sinon.spy(DeployerOption, "process");
+        
+        preloadDeploymentScript();
+    });
+
+    afterEach(async () => {
+        sinon.restore();
+        fs.removeSync('./contracts');
+        fs.removeSync('./deployment');
+    });
+    
+    after(async () => {
+        process.chdir(initialDir);
+        fs.removeSync(TEST_DIR);
+    });
+
+    function preloadDeploymentScript () {
+        fs.mkdirSync('./deployment');
+        fs.copyFileSync('../tests/cli-commands/mocks/deployment.js', './deployment/deployment.js');
+    }
+
+    it('Should initialize command properly', async () => {
+        assert(deployCommand instanceof Command);
+        assert(deployCommand.template == definition.template);
+        assert(deployCommand.description = definition.description);
+        assert(deployCommand.options == definition.options);
+        assert(deployCommand.subcommands.length == 0);
+    });
+
+    it('Should deploy when valid deployment folder is specified', async () => {
+        assert(await deployCommand.execute({ path: DEFAULT_PATH }));
+
+        sinon.assert.calledWith(pathOptionSpy, DEFAULT_PATH);
+        
+        let result = await pathOptionSpy.returnValues[0];
+        assert(result[0].fileName, 'deployment.js');
+        assert(typeof(result[0].deploy) == 'function');
+        assert(result[0].deploy.name, 'deploy');
+    });
+
+    it('Should throw when invalid deployment folder is specified', async () => {
+        assert(await deployCommand.execute({ path: INVALID_PATH }));
+
+        sinon.assert.calledWith(pathOptionSpy, INVALID_PATH);
+    });
+
+    it('Should not throw when deployment folder is empty', async () => {
```

Comment:
Should not throw when deployment folder is empty -> Should work with empty folder